ssh-audit.py

```diff
@@ -3713,14 +3720,55 @@ def audit(aconf: AuditConf, sshv: Optional[int] = None, print_target: bool = Fal
 
     return program_retval
 
+def algorithm_lookup(alg_names):       
+    if not alg_names:    
+        out.fail("[exception] Missing argument: alg_names")
+        sys.exit(1)
+        
+    alg_types = {'kex': 'key exchange algorithms', 
+                 'key': 'host-key algorithms',
+                 'mac': 'message authentication code algorithms',
+                 'enc': 'encryption algorithms (ciphers)'
+    }
+    
+    algorithms = alg_names.split(",")
+    adb = SSH2.KexDB.ALGORITHMS
+    algorithms_dict = {}
+    
+    for alg_type in alg_types:
+        for alg_name in algorithms:
+            try:
+                SSH2.KexDB.ALGORITHMS[alg_type][alg_name]
+                algorithms_dict.update( {alg_name : alg_type} ) 
+            except Exception:
+                pass
+
+    unknown_algorithms = []
+    padding = len(max(algorithms, key=len))
+        
+    for alg_type in alg_types:
+        algorithms_dict_filtered = {k:v for (k,v) in algorithms_dict.items() if alg_type in v}
+        
+        if len(algorithms_dict_filtered.items()) > 0:
+            title = alg_types.get(alg_type)
+            output_algorithms(title, adb, alg_type, algorithms_dict_filtered.keys(), unknown_algorithms, False, PROGRAM_RETVAL_GOOD, padding)
```

Comment:
The program should return the appropriate exit code when it ends.  So we should track the return value from `output_algorithms()`.

This can be done by initializing `retval = PROGRAM_RETVAL_GOOD`, then continually updating it with `retval = output_algorithms([...], retval, [...])`.  Then return `retval` to the caller.